patroni/postgresql/config.py

```diff
@@ -438,7 +448,9 @@ def write_postgresql_conf(self, configuration: Optional[CaseInsensitiveDict] = N
     def append_pg_hba(self, config: List[str]) -> bool:
         if not self.hba_file and not self._config.get('pg_hba'):
             with open(self._pg_hba_conf, 'a') as f:
+                self.set_file_permissions(self._pg_hba_conf)
                 f.write('\n{}\n'.format('\n'.join(config)))
+            self.set_file_permissions(self._pg_hba_conf)
```

Comment:
```suggestion
                f.write('\n{}\n'.format('\n'.join(config)))
            self.set_file_permissions(self._pg_hba_conf)
```

I guess we can remove one of the apparently redundant calls.